students/JaeKim/lesson03/assignment/test_basic_operations.py

```diff
@@ -0,0 +1,185 @@
+    """
+    setup_test_db()
+    add_customer(**OK_CUSTOMER)
+    valid_customer = TestCustomer.get(TestCustomer.customer_id == OK_CUSTOMER['customer_id'])
+
+    assert valid_customer.first_name == OK_CUSTOMER['name']
+    assert valid_customer.customer_id == OK_CUSTOMER['customer_id']
+    assert valid_customer.last_name == OK_CUSTOMER['lastname']
+    assert valid_customer.home_address == OK_CUSTOMER['home_address']
+    assert valid_customer.phone_number == OK_CUSTOMER['phone_number']
+    assert valid_customer.email_address == OK_CUSTOMER['email_address']
+    assert valid_customer.status == OK_CUSTOMER['status']
+    assert valid_customer.credit_limit == OK_CUSTOMER['credit_limit']
+
+    teardown_test_db()
+
+
+def test_credit_limit_float():
+    """
+    sets up a new db, adds customer, runs assertions and tears down db
+    """
+    setup_test_db()
+    bad_customer = dict(OK_CUSTOMER)
+    bad_customer['credit_limit'] = '$40'
+
+    with pytest.raises(ValueError):
+        add_customer(**bad_customer)
+
+    teardown_test_db()
+
+
+def test_get_customer():
+    """
+    sets up a new db, adds customer, runs assertions and tears down db
+    """
+    setup_test_db()
+    add_customer(**OK_CUSTOMER)
+    valid_customer = search_customer(OK_CUSTOMER['customer_id'], stage='dev')
+
+    assert valid_customer.first_name == OK_CUSTOMER['name']
+    assert valid_customer.customer_id == OK_CUSTOMER['customer_id']
+    assert valid_customer.last_name == OK_CUSTOMER['lastname']
+    assert valid_customer.home_address == OK_CUSTOMER['home_address']
+    assert valid_customer.phone_number == OK_CUSTOMER['phone_number']
+    assert valid_customer.email_address == OK_CUSTOMER['email_address']
+    assert valid_customer.status == OK_CUSTOMER['status']
+    assert valid_customer.credit_limit == OK_CUSTOMER['credit_limit']
+
+    teardown_test_db()
+
+
+def test_delete_customer():
+    """
+    sets up a new db, adds customer, runs assertions and tears down db
+    """
+    setup_test_db()
+
+    # Add and assert to confirm record exists before deleting
+    add_customer(**OK_CUSTOMER)
+    valid_customer = search_customer(OK_CUSTOMER['customer_id'], stage='dev')
+
+    assert valid_customer.first_name == OK_CUSTOMER['name']
+    assert valid_customer.customer_id == OK_CUSTOMER['customer_id']
+    assert valid_customer.last_name == OK_CUSTOMER['lastname']
+    assert valid_customer.home_address == OK_CUSTOMER['home_address']
+    assert valid_customer.phone_number == OK_CUSTOMER['phone_number']
+    assert valid_customer.email_address == OK_CUSTOMER['email_address']
+    assert valid_customer.status == OK_CUSTOMER['status']
+    assert valid_customer.credit_limit == OK_CUSTOMER['credit_limit']
+
+    delete_customer(OK_CUSTOMER['customer_id'], stage='dev')
+    valid_customer = search_customer(OK_CUSTOMER['customer_id'], stage='dev')
+
+    assert valid_customer is False
+
+    teardown_test_db()
```

Comment:
This always feel a bit weird, doing in one test what another has already tested. You don't have to make sure it is there after the add in this test. You tested that elsewhere; here you are testing delete, so just make sure it is gone after the delete. But, sometimes I do skip the add tests, since it gets tested so much in all of the other tests.